### PR TITLE
@: added external-init command

### PR DESCRIPTION
**Pull request by icasimpan**
@lmartinking

Here's a clean topic branch for adding new command "external-init". Feel free to add it as you please in your own time.

Thanks,
icasimpan

### CODE REVIEW

**Comment by icasimpan** [comment]
@lmartinking 

Feel free to delete this topic branch once you are done with the merge. This has been merged to my latest changes in branch "icasimpan-latest".

Thanks.

File: @ [hunk 1]
```diff
@@ -261,6 +262,30 @@ _external-add () {
...
+ origin=$1
+ [[ -f $origin ]] && origin=$( readlink -f $origin )
+ name=$( basename $origin )
+
+ echo $name
+
+ [[ -e $origin ]] && die "An external already exist by that name"
+
+ cat > $origin <<EOF
+  # @
```
Comment (lmartinking) [inline @]: I think this needs to be de-indented. Please check.

File: @ [hunk 2]
```diff
@@ -261,6 +262,30 @@ _external-add () {
...
+  # |external-skel||describe what external-skel do|
+  _external-skel () {
+   echo "body of external-skel here"
+  }
+EOF
+
+ sed -i "s/external-skel/$name/g" $origin
```
Comment (lmartinking) [inline @:291]: Why not reference "$name" in the here-doc? eg:

cat > Foo <<EOF
Home dir is $HOME
EOF

File: @ [hunk 3]
```diff
@@ -261,6 +262,30 @@ _external-add () {
...
+ sed -i "s/^  //" $origin
```
Comment (lmartinking) [inline @]: I see you are de-indenting it here. Just remove the indentation from above, as this is fragile.

File: @ (hunk 2 above)
Comment (icasimpan) [inline @:291]: Hi,

The reason I used $origin is that, I intend to be able to say:
  @ external-init ./external-path/dir1/dir2/mytoyfunction

If I would use $name, external-init will get confused...so for me, this is
a straightforward decision.

As for $name, its only use is to replace "external-skel" with its value.

On Thu, Aug 21, 2014 at 1:42 PM, Lucas Martin-King <notifications@github.com

> wrote:
> 
> In @:
> 
> > - [[ -f $origin ]] && origin=$( readlink -f $origin )
> > - name=$( basename $origin )
> >   +
> > - echo $name
> >   +
> > - [[ -e $origin ]] && die "An external already exist by that name"
> >   +
> > - cat > $origin <<EOF
> > -  # @
> > -  # |external-skel||describe what external-skel do|
> > -  _external-skel () {
> > -   echo "body of external-skel here"
> > -  }
> >   +EOF
> >   +
> > - sed -i "s/external-skel/$name/g" $origin
> 
> Why not reference "$name" in there here-doc? eg:
> 
> cat > Foo <<EOF
> Home dir is $HOME
> EOF
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/lmartinking/monkey-tail/pull/4/files#r16521428.

## 

Ismael Angelo A. Casimpan Jr.
ismael.angelo@casimpan.com

File: @ (hunk 3 above)
Comment (icasimpan) [inline @]: Ok, I'll de-indent heredoc and remove the de-indentation sed line...

On Thu, Aug 21, 2014 at 1:42 PM, Lucas Martin-King <notifications@github.com

> wrote:
> 
> In @:
> 
> > - name=$( basename $origin )
> >   +
> > - echo $name
> >   +
> > - [[ -e $origin ]] && die "An external already exist by that name"
> >   +
> > - cat > $origin <<EOF
> > -  # @
> > -  # |external-skel||describe what external-skel do|
> > -  _external-skel () {
> > -   echo "body of external-skel here"
> > -  }
> >   +EOF
> >   +
> > - sed -i "s/external-skel/$name/g" $origin
> > - sed -i "s/^  //" $origin
> 
> I see you are de-indenting it here. Just remove the indentation from
> above, as this is fragile.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/lmartinking/monkey-tail/pull/4/files#r16521446.

## 

Ismael Angelo A. Casimpan Jr.
ismael.angelo@casimpan.com

Comment (icasimpan) [inline @]: Hi,

De-indentation per your suggestion is implemented in my topic branch
(NEWCMD_external-init). See latest commit: 0b41c0b
https://github.com/icasimpan/monkey-tail/commit/0b41c0b492e90ca94653e59e0352ff8ef58bc5b8

Thanks.

On Thu, Aug 21, 2014 at 2:40 PM, Ismael Angelo Casimpan <
ismael.angelo@casimpan.com> wrote:

> Ok, I'll de-indent heredoc and remove the de-indentation sed line...
> 
> On Thu, Aug 21, 2014 at 1:42 PM, Lucas Martin-King <
> notifications@github.com> wrote:
> 
> > In @:
> > 
> > > - name=$( basename $origin )
> > >   +
> > > - echo $name
> > >   +
> > > - [[ -e $origin ]] && die "An external already exist by that name"
> > >   +
> > > - cat > $origin <<EOF
> > > -  # @
> > > -  # |external-skel||describe what external-skel do|
> > > -  _external-skel () {
> > > -   echo "body of external-skel here"
> > > -  }
> > >   +EOF
> > >   +
> > > - sed -i "s/external-skel/$name/g" $origin
> > > - sed -i "s/^  //" $origin
> > 
> > I see you are de-indenting it here. Just remove the indentation from
> > above, as this is fragile.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/lmartinking/monkey-tail/pull/4/files#r16521446.
> 
> ## 
> 
> Ismael Angelo A. Casimpan Jr.
> ismael.angelo@casimpan.com

## 

Ismael Angelo A. Casimpan Jr.
ismael.angelo@casimpan.com

**Comment by icasimpan** [comment]
Hi @lmartinking 

Please use the other contribution instead in my topic branch NEWCMD_external-init_FINAL.
It would only need 1 merge on your end.

Thanks,
icasimpan

**Pull request closed without merging**